pkg/stdout/capture.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package stdout
+
+import (
+	"bufio"
+	"context"
+	"log"
+	"os"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+func Capture(ctx context.Context, inner func() error) error {
+	r, _, cleanup := mustHijackStdOut()
```

Comment:
No need to capture `stdErr`?

Reply:
No, today we don't need.

Reply:
Would it not help identify any kind of failure in exporter if there is an issue in dcgm?

Reply:
No. Today it is not on the table for the change.